### PR TITLE
fix: [CI-5558]: CI Dashboard performance improvement

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37545)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/dashboard/CIOverviewDashboardServiceImpl.java [hunk 1]
```diff
@@ -71,11 +71,10 @@ public class CIOverviewDashboardServiceImpl implements CIOverviewDashboardServic
 
   // This ID comes, and should only come from internal source only. No need to use prepare statement here
   public String queryBuilderServiceTag(String queryIdCdTable) {
-    String selectStatusQuery =
-        "select service_name,tag,pipeline_execution_summary_cd_id from " + tableNameServiceAndInfra + " where ";
-
```
Comment (jamie-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/dashboard/CIOverviewDashboardServiceImpl.java:76]: I changed this to an exact match. I don't think there will be multiple ID here @harshjain12 let me know if I'm wrong

File: pipeline-service/service/src/main/java/io/harness/pms/migration/PipelineCoreTimeScaleMigrationDetails.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public List<Pair<Integer, Class<? extends NGMigration>>> getMigrations() {
         .add(Pair.of(8, AddInfrastructureIdentifierInServiceInfraInfoTable.class))
         .add(Pair.of(9, AddGitOpsEnabledInServiceInfraInfoTable.class))
         .add(Pair.of(10, AddInfrastructureNameInServiceInfraInfoTable.class))
+        .add(Pair.of(11, CreateServiceAndCIExecutionIndex.class))
```
Comment (naidu-harness) [inline pipeline-service/service/src/main/java/io/harness/pms/migration/PipelineCoreTimeScaleMigrationDetails.java:45]: we should remove these migrations from pipeline service and embed them in the CI manager.

File: pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution [hunk 3]
```diff
@@ -0,0 +1,4 @@
+BEGIN;
+CREATE INDEX IF NOT EXISTS service_infra_info_pipeline_execution_idx ON service_infra_info USING btree (pipeline_execution_summary_cd_id);
+CREATE INDEX IF NOT EXISTS pipeline_summery_ci_account_org_proj_idx ON pipeline_execution_summary_ci USING btree (accountid, orgidentifier, projectidentifier);
+COMMIT;
```
Comment (naidu-harness) [inline pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution:4]: ditto

**Review by naidu-harness — CHANGES_REQUESTED** [review]

File: pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution (hunk 3 above)
Comment (jamie-harness) [reply pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution]: ok let me figure this out. The service_infra_info table pipeline team owns it tho. Should it be in pipeline? @naidu-harness

Comment (naidu-harness) [reply pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution]: pipeline team does not own this table. it should CD

Comment (jamie-harness) [reply pipeline-service/service/src/main/resources/timescale/add_index_on_service_infra_info_pipe_execution]: got it. I will move it to CI. it's shared component of CI/CD

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37545)

**Review by naidu-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37545#-:-NCc4uGV0Ms9cpKdjO6b:bnfp4nl)* status: 11 file(s) to review.

File: 332-ci-manager/service/src/main/java/io/harness/ci/dashboard/CIOverviewDashboardServiceImpl.java [hunk 4]
```diff
@@ -71,11 +71,10 @@ public class CIOverviewDashboardServiceImpl implements CIOverviewDashboardServic
...
-    return String.format(
-        selectStatusQuery + "pipeline_execution_summary_cd_id in (%s) and service_name is not null;", queryIdCdTable);
+    return String.format("select service_name,tag,pipeline_execution_summary_cd_id from " + tableNameServiceAndInfra
+            + " where "
+            + "pipeline_execution_summary_cd_id=%s and service_name is not null;",
```
Comment (rutvijmehta-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/dashboard/CIOverviewDashboardServiceImpl.java:76]: Is this an internal query and safe from injection?

Comment (jamie-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/dashboard/CIOverviewDashboardServiceImpl.java:76]: yes it from a query result. And was previous populated by our code. It's a hash

**Review by rutvijmehta-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 10 of 10 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37545)* status: :shipit: complete! ✓

**Pull request merged**